project_group_setup.py

```diff
@@ -238,16 +253,24 @@ def main(args):
             project_id_index = identifier_index(identifier_list, "ospoolproject")
             if project_id_index != -1:
                 project_id = str(identifier_list[project_id_index]["Identifier"])
-                is_project = re.compile(OSPOOL_PROJECT_PREFIX_STR + "*").match(project_id) is not None
-            else:
-                is_project = False
+                if re.compile(OSPOOL_PROJECT_PREFIX_STR + "*").match(project_id) is not None:
+                    project_groups.add(gid)
 
             osggid_index = identifier_index(identifier_list, "osggid")
             if osggid_index != -1:
                 highest_osggid = max(highest_osggid, int(identifier_list[osggid_index]["Identifier"]))
-            elif is_project is True:
-                project_name = project_id.replace(OSPOOL_PROJECT_PREFIX_STR, "", 1).lower()
-                projects_to_assign_identifiers.append((gid, project_name,))
+
+            if gid in project_groups:
+                if osggid_index == -1:
+                    project_name = project_id.replace(OSPOOL_PROJECT_PREFIX_STR, "", 1).lower()
```

Comment:
I know this is what the script did previously but can we get this name from the "osgname" identifier (or whatever it's called) instead?

Reply:
I'm not quite sure if you mean the "osg group name" identifier for projects, or the future plan we talked about to transition the creation of project groups to using something like "project_UW_Swanson" as the CO Group name. If the former, this is the part of the code that determines what the value of the "osg group name" identifier should be for a project group. If the later, shouldn't we wait until all of the old-style "Is OSG Project" type projects are set up in LDAP before we change how we determine which groups are projects?

Reply:
> the future plan we talked about to transition the creation of project groups to using something like "project_UW_Swanson" as the CO Group name

This is a pipe dream for me right now so let's not make any decisions baked on my half-baked design idea :D 

> If the former, this is the part of the code that determines what the value of the "osg group name" identifier should be for a project group.

Ah, right. How about just taking it from the COManage group name instead? Potential brittleness of the string replacement worries me.

Reply:
![image](https://github.com/opensciencegrid/osg-comanage-project-usermap/assets/144060728/229734a3-1fe1-4c61-bd9c-1dfc1c804cbf)
Ok, the script now uses the CO group name to create the value of the OSG Group Name identifier.